teamcity: failed test: TestGenerateSubzoneSpans

The following tests appear to have failed on master (testrace): TestGenerateSubzoneSpans

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestGenerateSubzoneSpans).

[#1039998](https://teamcity.cockroachdb.com/viewLog.html?buildId=1039998):


```
TestGenerateSubzoneSpans
...m/cockroachdb/cockroach/pkg/util/log/clog.go:1255 +0x8b
/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x47c
/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1255 +0x8b
/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x47c
------- Stdout: -------
panic: locale is required for COLLATEDSTRING [recovered]

goroutine 3239 [running]:
testing.tRunner.func1(0xc4201781e0)
panic(0x3565c20, 0x42005f0)
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
panic(0x3565c20, 0x42005f0)
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.columnSemanticTypeToDatumType(0xc4219bb4d0, 0xa, 0x0, 0x0)
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.RandDatumWithNullChance(0xc420f3c270, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.RandDatum(0xc420f3c270, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.allPartitioningTests(0xc420f3c270, 0xc023b8f354d72136, 0xc421464280, 0xc421505f68)
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestGenerateSubzoneSpans(0xc4201781e0)
testing.tRunner(0xc4201781e0, 0x3b03590)
created by testing.(*T).Run

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4201780f0, 0x39e7c0d, 0x18, 0x3b03590, 0xc42072fc01)
testing.runTests.func1(0xc4201780f0)
testing.tRunner(0xc4201780f0, 0xc42072fd68)
testing.runTests(0xc4202e67a0, 0x55e0b20, 0x8, 0x8, 0xbef9d39071e44377)
testing.(*M).Run(0xc420774200, 0x0)
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestMain(0xc420774200)
main.main()

goroutine 6 [syscall]:
os/signal.signal_recv(0x959511)
os/signal.loop()
created by os/signal.init.0

goroutine 21 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0

goroutine 22 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0

goroutine 25 [select, locked to thread]:
runtime.gopark(0x3b08858, 0x0, 0x39c2c0c, 0x6, 0x18, 0x1)
runtime.selectgo(0xc42008ff50, 0xc4204b6300)
runtime.ensureSigM.func1()
runtime.goexit()

goroutine 10 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc4205c2600)
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0

goroutine 2714 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078e780)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 30 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078e280)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 31 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078e3c0)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 1799 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc4204d72c0)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 1052 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078ea00)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 1800 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc4204d7400)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 1053 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078eb40)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger

goroutine 2715 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc42078ec80)
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/sqlbase/testutils.go
+++ b/pkg/sql/sqlbase/testutils.go
@@ -198,8 +198,19 @@ func RandDatumWithNullChance(rng *rand.Rand, typ ColumnType, nullChance int) tre
 		return tree.DNull
 	case ColumnType_ARRAY:
 		if typ.ArrayContents == nil {
-			contentsTyp := RandColumnType(rng)
+			var contentsTyp ColumnType
+		LOOP:
+			for {
+				contentsTyp = RandColumnType(rng)
+				switch contentsTyp.SemanticType {
+				// Can't have an array of an array.
+				case ColumnType_ARRAY, ColumnType_JSONB:
+				default:
+					break LOOP
+				}
+			}
 			typ.ArrayContents = &contentsTyp.SemanticType
+			typ.Locale = contentsTyp.Locale
 		}
 		eltTyp := typ.elementColumnType()
 		datumType := columnSemanticTypeToDatumType(eltTyp, eltTyp.SemanticType)

